scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java

```diff
@@ -1071,7 +1092,8 @@ WorkflowMetadata getWorkflowMetadata(String workflowId, String correlationId) {
     }
 
     @VisibleForTesting
-    String lookupWorkflowIdFromTaskId(String taskId) {
+    @Cacheable(value = SHARD_ID_CACHE, key = "#workflowId")
```

Comment:
When we memoize the arguments are taken as the key, and is different here. Kindly check once.